src/CoreBundle/Service/RequestInfoService.php

```diff
@@ -129,6 +132,27 @@ public function isCmsTemplateEngineEditMode()
         return $this->isCmsTemplateEngineEditModeCache;
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function isPreviewMode()
+    {
+        if (null !== $this->isPreviewModeCache) {
+            return $this->isPreviewModeCache;
+        }
+
+        $request = $this->requestStack->getCurrentRequest();
+        if (null === $request) {
+            return false;
+        }
+
+        // todo: `__rpeviewmode` should be the only way to enable this. Refactor all places where the preview attribute is set as `preview` instead of `__previewmode'
+        $this->isPreviewModeCache = false === \TGlobal::IsCMSMode() &&
+            ('true' === $request->get('__previewmode') || 'true' === $request->get('preview'));
```

Comment:
Use InputFilterUtil here?

This is always a GET parameter? (could use the specific method then)

Reply:
We are not using the parameter in any way, we are just comparing it to a static string. So I don't see how an input filter would benefit us. Or do I miss something?

Yes, this is always a GET parameter. And I think it is very clear what is being checked here. I don't thing a separate method would benefit us much.

Reply:
Specific: I meant $request->query->get().

Reply:
ah, i see. Yes, that makes sense. done